Modules/StudyProgramme/classes/model/Progress/class.ilStudyProgrammeProgress.php

```diff
@@ -414,14 +414,17 @@ public function updateLastChange()
 	 * @throws ilException
 	 * @return $this
 	 */
-	public function setLastChange(ilDateTime $a_timestamp) : ilStudyProgrammeProgress
+	public function setLastChange(ilDateTime $a_timestamp, $force = false) : ilStudyProgrammeProgress
```

Comment:
Wozu brauchen wir diesen Force-Parameter? Der kommt mir seltsam vor.

Reply:
Damit ich beim laden des objekts nicht immer den last change auf den aktuellen timestamp setze beim reinschreiben der proerties.

Reply:
Das check ich nicht. Das schaltet doch nur den Check aus und an?

Reply:
Ja, das heisst du kannst es auf den wert setzen, der in der Datenbank steht und nicht auf den auktellen wert, der automatisch eingetragen wird, sobald du, z.B., die Punkte setzt. Siehe Funktion updateLastChange, die aufgerufen wird, wenn du punkte einträgst. Das muss ich aber machen, wenn ich das objekt lade. Würde ich dann versuchen den last-change wert aus der Datenbank einzutragen, würde es werfen...

Reply:
Ok. Mach den Check weg.

Reply:
Und damit auch das Parameter.

Reply:
Die Klasse ist ja nun im Grunde einfach eine Datenklasse (oder) und muss nichts mehr über externe Konsistenz aus den Businessregeln wissen. Der Check betrifft aber eine Businessregel. Einverstanden?

Reply:
Okely-dokely